pymatgen/transformations/standard_transformations.py

```diff
@@ -494,6 +495,111 @@ def inverse(self):
         return
 
 
+class RandomStructureTransformation(AbstractTransformation):
+    """
+    Transform a disordered structure into a given number of ordered random structures.
+    """
+
+    def __init__(self):
+        pass
+
+    def apply_transformation(self, structure: Structure, n_copies: int) -> list[Structure]:
+        """
+        For this transformation, the apply_transformation method will return
+        ordered random structure(s) from the given disordered structure.
+
+        Args:
+            structure: input structure
+            n_copies (int): number of copies of ordered random structures to be returned
+
+        Returns:
+            A list of ordered random structures based on the input disordered structure.
+        """
+        subl = structure.sublattices
+
+        # fill the sublattice sites with pure-element atoms
+        self.all_structures = []
```

Comment:
is it necessary to store the structures on the transformation instance? might balloon mem usage if people create many of these `RandomStructureTransformation`s.

Reply:
I used this syntax because there is an `is_one_to_many` property in the class; should it be `False` by default? (I see it is `True` for other classes)

Reply:
I think there's no connection between `is_one_to_many` (that part is good) and storing the transformed structures on the transformation instance?

Reply:
Could you elaborate more on this? I feel I'm not totally understanding. For example, the `apply_transformation` method of `OrderDisorderedTransformation` returns a list of structures as well (`self._all_structures`). What is the issue with this approach?

Reply:
Could you explain the need to store all structures for later? Why not just return them and let them be garbage collected if they go out of scope in the user's code? Currently they live as long as the `Transformation` instance which is not great given structures are notoriously mem hungry.

Reply:
I see. This feature is here simply because I'm copying what's been written for the other transformations. If you think it's not a good approach, then I can modify it in my code accordingly.